optimade/models/index_metadb.py

```diff
@@ -0,0 +1,54 @@
+from pydantic import Schema, BaseModel
+from typing import Union, Dict
+
+from .jsonapi import BaseResource
+from .baseinfo import BaseInfoAttributes, BaseInfoResource
+
+
+__all__ = (
+    "IndexInfoAttributes",
+    "RelatedChildResource",
+    "IndexRelationship",
+    "IndexInfoResource",
+)
+
+
+class IndexInfoAttributes(BaseInfoAttributes):
+    """Attributes for Base URL Info endpoint for an Index Meta-Database"""
+
+    is_index: bool = Schema(
+        default=True,
+        const=True,
+        description="If true, this is an index meta-database base URL (see section Index Meta-Database). "
+        "If this member is not provided, the client MUST assume this is not an index meta-database base URL "
+        "(i.e., the default is for is_index to be false).",
+    )
+
+
+class RelatedChildResource(BaseResource):
```

Comment:
Consider making this `BaseChildResource` and subclass this for `ChildResource`, or similar.

Reply:
Actually, consider subclassing `BaseResource` for all resources, before subclassing _that_ with the full resource, in order to more easily have something for the `relationships` field.

Reply:
This can pose some problems when wanting to hit all resources by typesetting for `Resource` or similar. Either `BaseResource` and `Resource` classes are subclassed for each type of resource, which will render the type of resource unassociated with each other directly, but one can still hit all "full" resources by going for `Resource` and hit _all_ resource classes by going for `BaseResource`, since `Resource` is a subclass of `BaseResource`.